### PR TITLE
Allow for an entity be passed into Link class 

### PR DESCRIPTION
**Pull request by thomasyu888**
https://sagebionetworks.jira.com/browse/SYNPY-494

### CODE REVIEW

**Comment by thomasyu888** [comment]
Can we pull this in?

File: tests/integration/integration_test_Entity.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def test_Entity():
 
     newfolder = Folder('Testing Folder', parent=project)
     newfolder = syn.store(newfolder)
-    link = Link(newfolder.id, parent = folder.id)
```
Comment (kdaily) [inline tests/integration/integration_test_Entity.py]: should this be removed, or another test created?

Comment (thomasyu888) [inline tests/integration/integration_test_Entity.py]: So this shouldn't matter anymore, as I fixed the Link entity to accept both Synapse ids or entities.

Comment (kdaily) [inline tests/integration/integration_test_Entity.py]: I meant should you be testing both (passing in ID or an entity object)? I'm fine with not just wondering what we do in other tests that create Synapse objects.

Comment (thomasyu888) [inline tests/integration/integration_test_Entity.py]: If you notice https://github.com/Sage-Bionetworks/synapsePythonClient/pull/420/files/83281f034455f150013c0de58177b6f89a09f3dd#diff-7c91124835c610ac5387325cc1920f6dR123

I do a link out to a file entity by using the id.  I passed in the newfolder entity to test the feature I added in the Link entity.

Comment (kdaily) [inline tests/integration/integration_test_Entity.py]: Aha - didn't expand far enough to see that. My bad!

Comment (thomasyu888) [inline tests/integration/integration_test_Entity.py]: Merge?

**Pull request merged**